src/docs/sphinx/buildGuide/Dependencies.rst

```diff
@@ -1,141 +1,128 @@
-- ``ENABLE_WARNINGS_AS_ERRORS``: GEOSX considers every warning as an error. When developing, you may face warnings however. You can modify this options (at your own risk) directly in the cmake scripts. Please understand that you won't be able to merge your code like this :)
+Third-party library dependencies
+================================
+
+GEOSX makes use of multiple third-party libraries (TPLs) and tools, some of which are mandatory and some optional.
+We only test against specific versions, and sometimes even require development snapshots (specific git commits).
+Not all of these guarantee backwards compatibility, so we strongly recommend building with these specific versions.
+
+List of third-party libraries
+-----------------------------
+
+The table below lists the dependencies with their specific versions and relevant CMake variables.
+Some of these libraries may have their own system prerequisites.
+For example, Doxygen depends on ``flex``/``bison``, ``latex``, ``ghostscript`` and ``graphviz``.
+
+============= ========== =========================== ============================= ========================================
+Name          Version    Enable option               Path variable                 Other options and comments
+============= ========== =========================== ============================= ========================================
+axom_         0.3.1      *mandatory*                 :code:`AXOM_DIR`
+Adiak_        0.2.0      :code:`ENABLE_CALIPER`      :code:`ADIAK_DIR`
+Caliper_      2.4.0      :code:`ENABLE_CALIPER`      :code:`CALIPER_DIR`
+conduit_      0.5.0      *mandatory*                 :code:`CONDUIT_DIR`
+CHAI_         2.2.2      *mandatory*                 :code:`CHAI_DIR`
+RAJA_         0.12.1     *mandatory*                 :code:`RAJA_DIR`
+hdf5_         1.10.5     *mandatory*                 :code:`HDF5_DIR`
+mathpresso_   2015-12-15 :code:`ENABLE_MATHPRESSO`   :code:`MATHPRESSO_DIR`
+pugixml_      1.8.0      *mandatory*                 :code:`PUGIXML_DIR`
+parmetis_     4.0.3      *mandatory* (with MPI)      :code:`PARMETIS_DIR`          built with 64-bit :code:`idx_t`
+suitesparse_  5.8.1      :code:`ENABLE_SUITESPARSE`  :code:`SUITESPARSE_DIR`
+superlu_dist_ 0f6efc3    :code:`ENABLE_SUPERLU_DIST` :code:`SUPERLU_DIST_DIR`
+hypre_        2186a8f    :code:`ENABLE_HYPRE`        :code:`HYPRE_DIR`
+PETSc_        3.13.0     :code:`ENABLE_PETSC`        :code:`PETSC_DIR`
+Trilinos_     12.18.1    :code:`ENABLE_TRILINOS`     :code:`TRILINOS_DIR`
+silo_         4.10.3     *mandatory*                 :code:`SILO_DIR`
+VTK_          9.0.0-rc3  :code:`ENABLE_VTK`          :code:`VTK_DIR`               only a small subset of modules required
+doxygen_      1.8.20     :code:`ENABLE_DOXYGEN`      :code:`DOXYGEN_EXECUTABLE`
+sphinx_       1.8.20     :code:`ENABLE_SPHINX`       :code:`SPHINX_EXECUTABLE`
+uncrustify_   401a409    :code:`ENABLE_UNCRUSTIFY`   :code:`UNCRUSTIFY_EXECUTABLE`
```

Comment:
I'm afraid the versions will never be maintained.

Reply:
Yeah... unless we're super vigilant on all and any `thirdPartyLibs` PRs. I just thought, since there is a way for users to provide dependencies separately (without `thirdPartyLibs`), it'd be good to have required versions listed in the documentation. I see no problem removing it though.